trace/trace.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trace
+
+import (
+	"context"
+)
+
+// executableTraceHookKey is a value key used to retrieve the ExecutableTrace
+// from Context.
+type executableTraceContextKey struct{}
+
+// ExecutableTrace is a set of hooks used to trace the execution of the binary
+// executable used with NativeStore. Any particular hook may be nil.
+type ExecutableTrace struct {
+	// ExecuteStart is called before the execution of the executable. The
+	// executableName parameter is the name of the credential helper executable
+	// used with NativeStore. The action parameter is one of "store", "get" and
```

Comment:
Just curious, what would happen if I passed a string other than [store, get, erase]? 
(Personally, I'd prefer to make variable like `action` to enum instead of a string as it only takes 3 valid values.)

Reply:
`ExecuteStart` is for logging purpose, and is always invoked by `oras`. If a user hook receives action other than [store, get, erase]. It's probably a programming bug.

Reply:
For your information, if you pass a string other than [store, get, erase], the behavior is up to the executable. For docker credential helper executables, an error will be thrown. @Two-Hearts